homeassistant/components/home_connect/switch.py

```diff
@@ -295,48 +294,34 @@ async def async_turn_off(self, **kwargs: Any) -> None:
                     **get_dict_from_home_connect_error(err),
                 },
             ) from err
-        self.async_entity_update()
 
-    async def async_update(self) -> None:
-        """Update the switch's status."""
-        state = self.device.appliance.status.get(BSH_ACTIVE_PROGRAM, {})
-        if state.get(ATTR_VALUE) == self.program_name:
+    def update_native_value(self) -> None:
+        """Update the switch's status based on if the program related to this entity is currently active."""
+        event = self.appliance.events.get(EventKey.BSH_COMMON_ROOT_ACTIVE_PROGRAM)
+        if event and event.value == self.program.key:
             self._attr_is_on = True
         else:
             self._attr_is_on = False
-        _LOGGER.debug("Updated, new state: %s", self._attr_is_on)
 
 
 class HomeConnectPowerSwitch(HomeConnectEntity, SwitchEntity):
     """Power switch class for Home Connect."""
 
     power_off_state: str | None
```

Comment:
Side note: I suggest we initialize the attribute to a sentinel value instead. It's bad practice in Python to have uninitialized attributes after the instance has been created unless there are some specific circumstances that require that.

https://github.com/home-assistant/core/blob/1face8df565e4ad16281bcfcf8dc7a280a15c203/homeassistant/helpers/typing.py#L31-L37